### PR TITLE
Add `disabled` prop to SegmentedControl

### PR DESCRIPTION
**Pull request by LucasLemanowicz**
`disabled={false}`:
Screen Shot 2022-01-23 at 3 17 44 PM

`disabled={true}`:
Screen Shot 2022-01-23 at 3 17 31 PM

### CODE REVIEW

File: src/mantine-core/src/components/SegmentedControl/SegmentedControl.styles.ts [hunk 1]
```diff
@@ -131,6 +131,12 @@ export default createStyles(
         },
       },
 
+      disabled: {
+        '&, &:hover': {
+          color: theme.colors.gray[5],
```
Comment (LucasLemanowicz) [inline src/mantine-core/src/components/SegmentedControl/SegmentedControl.styles.ts:136]: @rtivital not sure if there's a more generic way here to support custom themes?

**Comment by rtivital** [comment]
`disabled` class is missing in Styles API description

**Comment by rtivital** [comment]
I think you've forgotten about dark color scheme, disabled component does not differ much and in some places has even more contrast than not disabled one

**Comment by LucasLemanowicz** [comment]
ok, which one of these do you prefer?

Enabled SegmentControl in dark mode:
segment-dark

Disabled (Gray 5):
segment-dark-disabled-gray5

Disabled (Gray 6):
segment-dark-disabled-gray6

Disabled (Gray 7):
segment-dark-disabled-gray7

**Comment by rtivital** [comment]
@LucasLemanowicz dark theme should not use gray colors, it can use only `theme.colors.dark`

**Comment by rtivital** [comment]
I'll finish it myself, don't want to bother you too much, thanks for a PR!

**Pull request merged**